Updated logic in Plan.num_answered_questions to allow for Phase context

briri opened the pull request:
Fixes #2095 .

Addresses an issue with the progress bar 'answered questions` count when a plan is for a template with multiple phases. Updated the `num_answered_questions` method to accept a Phase so that it can return either the total number of questions answered for the entire plan or a specific section.

xsrust commented on app/models/plan.rb:
```diff
@@ -486,11 +486,26 @@ def authors
+      # If a phase was specified then return the number of
+      # answered questions within the context of the phase
+      question_ids = phase.sections.map do |s|
+        s.questions.pluck('questions.id')
```
I think that the query would be more efficient if we defined the relationship between Phase and questions, or phase and answers, probably both.

xsrust commented on app/models/plan.rb:
```diff
@@ -486,11 +486,26 @@ def authors
+            .includes(:question_options, question: :question_format)
+            .where(question_id: question_ids.flatten)
+            .to_a
+            .sum { |answer| answer.is_valid? ? 1 : 0 }
```
If the relationship between a phase and answers is defined, 

`phase.answers.where(plan_id: self.id).includes(:question_options, question: :question_format).to_a.sum`...

xsrust commented on app/models/plan.rb:
```diff
@@ -486,11 +486,26 @@ def authors
+            .sum { |answer| answer.is_valid? ? 1 : 0 }
+    else
+      # Return the total number of answered questions across all phases
+      Answer.where(id: answers.map(&:id))
```
Think it would be simpler to replace this with `self.answers`,  as we're finding the id's and then re-finding those answers.

Pre-fetching should still work on the association

briri commented:
makes sense @xsrust. Will update the model's relationships

briri commented:
Updated the models by adding the association shortcut as suggested. Also updated the name of `is_valid?` to `answered?` which seems more Rails-ish.
